File: Day-13-of-30-Days/Solution.js
function calculateVowels(input) {
    let vowelCounts = {};
    const vowels = "aeiou";

    for (let i = 0; i < vowels.length; i++) {
        let vowel = vowels[i];
        vowelCounts[vowel] = 0;
    }

    input = input.toLowerCase();

    for (let i = 0; i < input.length; i++) {
        let c = input[i];
        if (vowels.includes(c)) {
            vowelCounts[c]++;
        }
    }

    return vowelCounts;
}


console.log(calculateVowels("Hello World"));
console.log(calculateVowels("Try"));

File: Day-13-of-30-Days/Solution.java
import java.util.*;
public class Solution {
    public static Map<Character, Integer> calculateVowels(String input) {
        // Initialize a map to store vowel counts
        Map<Character, Integer> vowelCounts = new HashMap<>();
        String vowels = "aeiou";

        for (char vowel : vowels.toCharArray()) {
            vowelCounts.put(vowel, 0);
        }
        input = input.toLowerCase();

        // Iterate through each character in the input string
        for (char c : input.toCharArray()) {
            if (vowels.contains(Character.toString(c))) {
                vowelCounts.put(c, vowelCounts.get(c) + 1);
            }
        }

        return vowelCounts;
    }

    public static void main(String[] args) {
        System.out.println(calculateVowels("Hello World"));
        System.out.println(calculateVowels("Try"));
    }
}
